velox/type/TimestampConversion.h

```diff
@@ -44,6 +44,26 @@ constexpr const int32_t kMaxYear{292278994};
 constexpr const int32_t kYearInterval{400};
 constexpr const int32_t kDaysPerYearInterval{146097};
 
+// Enum to dictate parsing modes for date strings.
+//
+// kStrict: For date string conversion, align with DuckDB's implementation.
+//
+// kNonStrict: For timestamp string conversion, align with DuckDB's
+// implementation.
+//
+// kStandardCast: Strictly processes dates in the [+-](YYYY-MM-DD) format.
+// Align with Presto casting conventions.
+//
+// kNonStandardCast: Like standard but permits missing day/month and allows
+// trailing 'T' or spaces. Align with Spark SQL casting conventions.
+enum class ParseMode {
+  kStrict,
```

Comment:
These names are not very descriptive. I wonder if we could come up with names that are more self-explanatory. What do you think?

Reply:
I do agree these names aren't great. Though perhaps it would be better to look into this separately, in conjunction with investigating the necessity of `kStrict` (https://github.com/facebookincubator/velox/pull/9357#discussion_r1561700407)? That will require identifying the purpose of each mode anyway and seeing if some can be removed/merged, after which it may be easier to rename them appropriately.